platform/android/llapp_main.lua

```diff
@@ -26,5 +25,5 @@ A.execute("chmod", "755", "./zsync")
 C.setenv("TESSDATA_PREFIX", "/sdcard/koreader/data", 1)
 
 -- create fake command-line arguments
-arg = {"-d", file or "/sdcard"}
+arg = {file or "/sdcard"}
```

Comment:
On Android you can't easily adjust the command line arguments like on other platforms, so I'm not convinced this is a good change.

Reply:
I think the intent is to take advantage of the debug toggle done in reader.lua instead of hard-coding this here?

Reply:
My point is that if you can't reach that toggle, it's a minor inconvenience on other platforms since you can launch the program with -d manually. For Android, by contrast, you need an Android dev environment.

Reply:
Although now that I think about it you can easily add the option to settings.lua now, so never mind I suppose.

But I very much don't like it regardless. We know when it's a debug flavor, right? So please enable it by default for a debug build. ;-)

Reply:
Eh, fair point.

Not necessarily a fan of enforcing debug logging by default, though, but I see what you mean.

Reply:
Oh, right, making it depend on the build flavor does take care of the situation in a more elegant manner :).

Reply:
Errr, AFAICT it just hides the blitbuffer code a bit. It doesn't affect to luajit-launcher stuff that is calling androig.log directly.

The only difference between debug and release builds should be that in release builds we don't log debug statements (aka Log.d, which is currently unused in both frontend and launcher)

Reply:
I vehemently disagree. I want my debug builds to fail hard on asserts. You could distinguish between different types of debug builds, but a build with debugging symbols combined with NDEBUG false barely serves a purpose in my eyes, if any. You could always implement that distinction if you want. ;-)

However, that's neither here nor there. This change removes debug statements for any build type regardless of assert behavior. I spent a lot of time and effort on making `./kodev run android` as painless as possible. I'm not responsible for Android being slightly overenthusiastic about debug logging, but I would've nevertheless made sure it behaved that way. An important aspect of painless is the `-d` flag like for the desktop emulator:

https://github.com/koreader/koreader/blob/7396abebf26ecbb3a8cf45f4ee4dcb61cb788287/kodev#L663

I don't much care how it's done, but I strongly object to anything that makes Android even the tiniest bit harder. Because as stupid as it might sound, that extra bit of friction, no matter how small, directly translates into less or no Android testing on my end.

Reply:
> You could always implement that distinction if you want. ;-)

Nah, I really don't care.